src/ILCompiler.Compiler/src/Compiler/CompilerTypeSystemContext.cs

```diff
@@ -238,7 +238,7 @@ private EcmaModule AddModule(string filePath, string expectedSimpleName)
                     ModuleData actualModuleData = _simpleNameHashtable.AddOrGetExisting(moduleData);
                     if (actualModuleData != moduleData)
                     {
-                        if (actualModuleData.FilePath != filePath)
+                        if (actualModuleData.FilePath != filePath && actualModuleData.SimpleName != "mscorlib")
```

Comment:
@MichalStrehovsky The only reason I used a hack here is because I don't know how to fix this for real. If you can suggest a more robust fix, I will gladly use that instead.